### PR TITLE
LOK-2478 : Syslog message from camel route to Syslog SinkModule

### PR DESCRIPTION
**Pull request by MusaidAli**
## Description
Syslog message from camel route to Syslog SinkModule

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2478
- 

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java [hunk 1]
```diff
@@ -76,7 +106,36 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
                                 (InetSocketAddress) exchange.getIn().getHeader(NettyConstants.NETTY_REMOTE_ADDRESS);
 
                         LOG.info(source.getHostName());
-                        LOG.info("Message received : " + exchange.getIn().getBody());
+                        LOG.info("Message received : {}", exchange.getIn().getBody());
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java:109]: Reduce this to level debug or even trace

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java [hunk 2]
```diff
@@ -76,7 +106,36 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+
+                        try {
+                            String content = buffer.toString(StandardCharsets.UTF_8);
+                            LOG.info("content{}", content);
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java:113]: Same. remove or reduce level

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java [hunk 3]
```diff
@@ -76,7 +106,36 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+                            SyslogMessage m_message = syslogParser.parse(content);
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java:114]: Still the same: broken naming and doing full-cycle parsing for nothing

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java [hunk 4]
```diff
@@ -76,7 +106,36 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+
+                            Identity identity1 = Identity.newBuilder()
+                                    .setSystemId("SyslogID")
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java:117]: Again, what is this ID?

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java [hunk 5]
```diff
@@ -76,7 +106,36 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+                                    .build();
+                            SyslogMessageDTO message = SyslogMessageDTO.newBuilder()
+                                    .setRawMessage(ByteString.copyFrom(
+                                            m_message.toString().getBytes(StandardCharsets.UTF_8)))
+                                    .build();
+
+                            SyslogMessageLogDTO dto = SyslogMessageLogDTO.newBuilder()
+                                    .setSyslogAddress(source.getHostName())
+                                    .setIdentity(identity1)
+                                    .setSyslogAddress(source.getHostName())
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java:127]: Setting address twice

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java [hunk 6]
```diff
@@ -76,7 +106,36 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+                                    .addSyslogMessageDTO(message)
+                                    .build();
+                            getMessageDispatcher().send(dto);
+
+                            LOG.info("Message sent");
+
+                        } catch (final Exception ex) {
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java:134]: See old comments

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java [hunk 7]
```diff
@@ -76,7 +106,36 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+
+                            SyslogMessageLogDTO dto = SyslogMessageLogDTO.newBuilder()
+                                    .setSyslogAddress(source.getHostName())
+                                    .setIdentity(identity1)
+                                    .setSyslogAddress(source.getHostName())
+                                    .addSyslogMessageDTO(message)
+                                    .build();
+                            getMessageDispatcher().send(dto);
+
+                            LOG.info("Message sent");
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java:132]: See old comments

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogMessage.java [hunk 8]
```diff
@@ -30,11 +31,23 @@
...
+
+    private Integer year;
+    private Integer month;
+    private Integer dayOfMonth;
+    private Integer hourOfDay;
+    private Integer minute;
+    private Integer second;
+    private Integer millisecond;
+    // private ZoneId zoneId;
+    // will add
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogMessage.java:47]: See old comments

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogParser.java [hunk 9]
```diff
@@ -21,37 +21,207 @@
...
 public class SyslogParser {
     private static final Logger LOG = LoggerFactory.getLogger(SyslogParser.class);
     private static final Pattern m_messageIdPattern = Pattern.compile("^((\\S+):\\s*)");
-    // date pattern has been updated to support space at start and end of
-    // the message due to which the date match failed and current system date
-    // used to be passed which cause 3ms more time in parsing message
     private static final Pattern m_datePattern = Pattern.compile("^\\s*((\\d\\d\\d\\d-\\d\\d-\\d\\d)\\s*)");
     private static final Pattern m_oldDatePattern =
             Pattern.compile("^\\s*(\\S\\S\\S\\s+\\d{1,2}\\s+\\d\\d:\\d\\d:\\d\\d)\\s+");
+    private static final String datePattern = "((19|20)\\d{2})-([1-9]|0[1-9]|1[0-2])-(0[1-9]|[1-9]|[12][0-9]|3[01])";
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogParser.java:45]: See old comments

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogParser.java [hunk 10]
```diff
@@ -21,37 +21,207 @@
...
+                LOG.trace("Found message '{}'", matchedMessage);
+            }
+
+            syslogMessage.setHostname(m.group(m_matchingGroupHost));
+
+            message = matchedMessage;
+        } else {
+            LOG.info("Regexp not matched: {}", message);
+            return null;
+        }*/
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogParser.java:153]: See old comments

File: minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 11]
```diff
@@ -1,28 +1,60 @@
...
+
+    <cm:property-placeholder id="syslogProperties"  persistent-id="org.opennms.horizon.minion.syslog.listener.syslog" update-strategy="reload">
+        <cm:default-properties>
+            <cm:property name="syslog.listen.interface" value="0.0.0.0" />
+            <cm:property name="syslog.listen.port" value="20514" />
+            <cm:property name="syslog.threads" value="10" />
+            <cm:property name="syslog.queue.size" value="10000" />
+            <cm:property name="syslog.batch.size" value="1000" />
+            <cm:property name="syslog.batch.interval" value="500" />
+            <cm:property name="syslog.forwardingPattern" value="^.*\s*(19|20)\d\d([-/.])(0[1-9]|1[012])\2(0[1-9]|[12][0-9]|3[01])(\s+)(\S+)(\s)(\S.+)" />
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml:24]: What does this forwarding pattern do and why does it need to be configurable?

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/java/org/opennms/horizon/shared/grpc/syslog/contract/mapper/impl/TenantLocationSpecificSyslogMessageDTOMapperImpl.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+ * language governing permissions and limitations under the
+ * License.
+ */
+package org.opennms.horizon.shared.grpc.syslog.contract.mapper.impl;
+
+import org.opennms.horizon.grpc.syslog.contract.SyslogMessageLogDTO;
+import org.opennms.horizon.grpc.syslog.contract.TenantLocationSpecificSyslogLogDTO;
+import org.opennms.horizon.shared.grpc.syslog.contract.mapper.TenantLocationSpecificSyslogMessageDTOMapper;
+
+public class TenantLocationSpecificSyslogMessageDTOMapperImpl implements TenantLocationSpecificSyslogMessageDTOMapper {
```
Comment (fooker) [inline shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/java/org/opennms/horizon/shared/grpc/syslog/contract/mapper/impl/TenantLocationSpecificSyslogMessageDTOMapperImpl.java:28]: Why is this not generated by mapstruct?

**Review by fooker — CHANGES_REQUESTED** [review]

File: minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml (hunk 11 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/resources/OSGI-INF/blueprint/blueprint.xml:24]: it was configurable in classic ,and there is todo task for it ,currently not handled.

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/java/org/opennms/horizon/shared/grpc/syslog/contract/mapper/impl/TenantLocationSpecificSyslogMessageDTOMapperImpl.java (hunk 12 above)
Comment (MusaidAli) [reply shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/java/org/opennms/horizon/shared/grpc/syslog/contract/mapper/impl/TenantLocationSpecificSyslogMessageDTOMapperImpl.java:28]: same was like for traps ,we follow traps

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java (hunk 1 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java]: reduced to log.trace.

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java (hunk 2 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java]: reduced to log.trace.

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java (hunk 5 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java]: removed ,other

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogParser.java (hunk 9 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogParser.java:45]: where are old comment ,i didn't find would you share here.thanks

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/java/org/opennms/horizon/shared/grpc/syslog/contract/mapper/impl/TenantLocationSpecificSyslogMessageDTOMapperImpl.java (hunk 12 above)
Comment (fooker) [reply shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/java/org/opennms/horizon/shared/grpc/syslog/contract/mapper/impl/TenantLocationSpecificSyslogMessageDTOMapperImpl.java:28]: Just you've copied crappy code does not mean you do not have to correct it. Please verify if it is possible to use mapstruct here.

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogParser.java (hunk 9 above)
Comment (fooker) [reply minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/parser/SyslogParser.java:45]: Took me a whole 30 secs to find these...

- https://github.com/OpenNMS-Cloud/lokahi/pull/1885/files/1b299134a109ce1c36d7105be38474a17c3a1b58#r1611367579
- https://github.com/OpenNMS-Cloud/lokahi/pull/1885/files/1b299134a109ce1c36d7105be38474a17c3a1b58#r1611369657
- https://github.com/OpenNMS-Cloud/lokahi/pull/1885/files/1b299134a109ce1c36d7105be38474a17c3a1b58#r1611372598

**Review by fooker — CHANGES_REQUESTED** [review]

**Pull request closed without merging**